Makefile

```diff
@@ -1,7 +1,6 @@
-CC=gcc
 CFLAGS=-O2 -Wall -Wpedantic -Wextra
 
-main: main.c
+main: main.o
```

Comment:
this dependency is also optional. Makefile knows that binary depends on object files.